contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -488,9 +488,12 @@ public static CFStringRef createCFString(String s) {
          */
         public String stringValue() {
             CFIndex length = INSTANCE.CFStringGetLength(this);
+            if (length.longValue() == 0) {
+                return "";
+            }
             CFIndex maxSize = INSTANCE.CFStringGetMaximumSizeForEncoding(length, kCFStringEncodingUTF8);
             if (maxSize.intValue() == kCFNotFound) {
-                return null;
+                throw new StringIndexOutOfBoundsException("CFString maximum number of bytes exceeds LONG_MAX.");
```

Comment:
Can you also throw another exception below for the case [CFStringGetCString](https://developer.apple.com/documentation/corefoundation/1542721-cfstringgetcstring?language=objc) returns 0:
> false if the conversion fails or the provided buffer is too small.

Reply:
First of all I would like to reforce that I'm not a C/C++ programer and my opnion is based in example and sources that I found on projects using CFStringGetCString :).
On theses projects/examples they never throw errors if this function returns 0, someones free allocated buffer (That's I think JNA do on GC for created Memory instance).
I could find only two cenarios, returning null or ignore result and read created buffer (garbage???).

Reply:
Yes, JNA will release the buffer on GC.   The docs indicate failure on two possibilities: the buffer to small (which should in theory be impossible given the steps preceding it) or if conversion fails (I can imagine possibly some malformed unicode).

So an exception equating to 'unicode conversion failed' types of errors should be appropriate here, probably an `IllegalArgumentException` is appropriate.